src/services/SelfTest.ts

```diff
@@ -85,70 +110,107 @@ export default class SelfTest {
           ws.on('error', (err: any) => {
             log.error('Error on connection for websockets', err)
             this.config.selfTestSuccess = false
-            process.exit(1)
+            this.wsSuccess = false
           })
         })
 
         const serverPromise = new Promise(async resolve => {
           try {
             const serverRes = await axios.get(`https://${response.manifestHash}.${url.host}`)
             const serverCheck = serverRes.data
-            log.debug('Pod upload succeeded', serverCheck)
+            log.debug('Pod HTTP request succeeded', serverCheck)
             if (serverCheck.imageUploaded) {
               // resolve promise
-              this.uploadSuccess = true
+              this.httpSuccess = true
               resolve()
             }
           } catch (err) {
-            log.error('Test contract not running', err)
+            log.error('Test pod not reachable', err)
             this.config.selfTestSuccess = false
-            process.exit(1)
+            this.httpSuccess = false
           }
         })
-        const testPromises = await Promise.all([serverPromise, webSocketPromise])
-          // Test that none of these promises are hanging for more than 60 seconds
-          // Timeout is set so that the contract has time to be pulled.
-        Promise.race([
-          testPromises,
-          new Promise((resolve, reject) => {
-            let timeout = setTimeout(function () {
-              clearTimeout(timeout)
-              reject(new Error('Could not listen to server or websocket due to timeout'))
-            }, 60000)
-          })
-        ])
-        .then(() => {
-          this.config.selfTestSuccess = (this.uploadSuccess && this.wsSuccess)
-          if (this.config.selfTestSuccess) {
-            log.info('Codius host passed self test! Ready to accept contracts')
-          } else {
-            log.error(`One or more self tests failed during startup. Unable to accept contracts. Uploads=${this.uploadSuccess}, Websockets=${this.wsSuccess}`)
-            process.exit(1)
-          }
+
+        const serverTimeoutPromise = new Promise((resolve, reject) => {
+          let timeout = setTimeout(function () {
+            clearTimeout(timeout)
+            reject(new Error('could not connect to pod due to timeout'))
+          }, 10000)
         })
-        .catch(err => {
-          this.config.selfTestSuccess = false
-          log.error('Error occurred when accessing self-test contract ', err)
-          process.exit(1)
+
+        const wsTimeoutPromise = new Promise((resolve, reject) => {
+          let timeout = setTimeout(function () {
+            clearTimeout(timeout)
+            reject(new Error('could not upload pod to server due to websocket timeout'))
+          }, 10000)
         })
+
+        for (let i = 0; i < this.testConfig.retryCount; i++) {
```

Comment:
Should this loop 5 times for both sequential tests. 
I think the Http should be tried 5 times, then the websocket one just once. Is there any value in trying the websockets many times.